semiphemeral: init at 0.6

amanjeev opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Adds python package [Semiphemeral](https://github.com/micahflee/semiphemeral). 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 120619` at 34ebeb4f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.semiphemeral</li>
    <li>python39Packages.semiphemeral</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/semiphemeral/default.nix:26:0:
```
   |
26 |     description = "Automatically delete your old tweets, except for the ones you want to keep";
   | ^
```
</li>
  </ul>
</details>

dotlambda commented on pkgs/development/python-modules/semiphemeral/default.nix:
```diff
@@ -0,0 +1,31 @@
+    license = licenses.mit;
+    maintainers = with maintainers; [ amanjeev ];
+  };
+}
```
add a final `\n`

dotlambda commented on pkgs/development/python-modules/semiphemeral/default.nix:
```diff
@@ -0,0 +1,31 @@
+, colorama
+}:
+
+buildPythonPackage rec {
```
```suggestion
buildPythonApplication rec {
```

dotlambda commented on pkgs/development/python-modules/semiphemeral/default.nix:
```diff
@@ -0,0 +1,31 @@
+    pname = "semiphemeral";
+    version = "0.6";
+
+    src = fetchPypi {
```
```suggestion
  src = fetchPypi {
```

dotlambda commented on pkgs/development/python-modules/semiphemeral/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  pythonImportsCheck = [ "semiphemeral" ];
+
+  buildInputs = [ click sqlalchemy flask tweepy colorama ];
```
```suggestion
```

dotlambda reviewed (changes requested):
This isn't a Python module (no `import semiphemeral`), so put it in `pkgs/tools/...`.

Also, you need exactly two commits:
* `maintainers: add amanjeev` and
* `semiphemeral: init at 0.6`

amanjeev committed:
maintainers: add Amanjeev Sethi

amanjeev committed:
semiphemeral: init at 0.6

semiphemeral: init at 0.6

semiphemeral: init at 0.6

semiphemeral: init at 0.6

semiphemeral: fix formatting

semiphemeral: refactor for flexibility

semiphemeral: remove extraneous inherit

semiphemeral: no check since upstream has no tests

semiphemeral: refactor move to misc instead of social/twitter

semiphemeral: fix formatting

dotlambda commented on pkgs/tools/social/twitter/semiphemeral/default.nix:
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, click
+, sqlalchemy
+, flask
+, tweepy
+, colorama
+}:
```
```suggestion
{ lib
, python3
}:
```

amanjeev replied:
thank you, this is really a lot of experience building

dotlambda replied:
The reason we do this is so that it's easier to override Python packages. That's done by using
```nix
semiphemeral.override {
  python3 = python3.override {
    packagesOverrides = ...;
  };
}
```

amanjeev replied:
I am not sure if I understand.  This does not build. 

```
error: undefined variable 'click' at /home/aj/proj/nix-nixos/nixpkgs/pkgs/tools/social/twitter/semiphemeral/default.nix:16:29
```

dotlambda replied:
See the newest review for how to do it.

dotlambda commented on pkgs/tools/social/twitter/semiphemeral/default.nix:
```diff
@@ -0,0 +1,30 @@
+, colorama
+}:
+
+buildPythonApplication rec {
```
```suggestion
python3.pkgs.buildPythonApplication rec {
```

dotlambda commented on pkgs/tools/social/twitter/semiphemeral/default.nix:
```diff
@@ -0,0 +1,30 @@
+}:
+
+buildPythonApplication rec {
+    pname = "semiphemeral";
+    version = "0.6";
```
```suggestion
  pname = "semiphemeral";
  version = "0.6";
```

dotlambda reviewed (changes requested):
Please also address @r-rmcgibbo's suggestion.

dotlambda commented on pkgs/tools/social/twitter/semiphemeral/default.nix:
```diff
@@ -0,0 +1,30 @@
+
+  pythonImportsCheck = [ "semiphemeral" ];
+
+  propagatedBuildInputs = [ click sqlalchemy flask tweepy colorama ];
```
```suggestion
  propagatedBuildInputs = with python3.pkgs; [ click sqlalchemy flask tweepy colorama ];
```

dotlambda commented on pkgs/tools/social/twitter/semiphemeral/default.nix:
```diff
@@ -0,0 +1,30 @@
+  pname = "semiphemeral";
+  version = "0.6";
+
+  src = fetchPypi {
```
```suggestion
  src = python3.pkgs.fetchPypi {
```

amanjeev commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25795,6 +25795,17 @@ in
 
+  semiphemeral = callPackage ../tools/social/twitter/semiphemeral {
+    inherit (python3Packages)
+    buildPythonApplication
+    fetchPypi
+    click
+    sqlalchemy
+    flask
+    tweepy
+    colorama;
```
I had to remove these too! Otherwise you get the error 

```
error: anonymous function at /home/aj/proj/nix-nixos/nixpkgs/pkgs/tools/social/twitter/semiphemeral/default.nix:1:1 called with unexpected argument 'buildPythonApplication', at /home/aj/proj/nix-nixos/nixpkgs/lib/customisation.nix:69:16
```

dotlambda replied:
Yes, you do have to remove them.

dotlambda commented:
Mind the two commits.

dotlambda commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25795,6 +25795,10 @@ in
 
   secretscanner = callPackage ../tools/security/secretscanner { };
 
+  semiphemeral = callPackage ../tools/social/twitter/semiphemeral {
+    inherit (python3Packages);
+  };
```
```suggestion
  semiphemeral = callPackage ../tools/social/twitter/semiphemeral { };
```

amanjeev commented:
Would it be ok to squash the commits before merging or should I squash and push again?

dotlambda reviewed (changes requested):
You should set `doCheck = false` and add a comment since upstream seems to have no tests.

dotlambda commented:
> Would it be ok to squash the commits before merging or should I squash and push again?

I can only squash multiple commits into a single one when merging.

amanjeev commented:
> Mind the two commits.

Yea I am can squash n -1 on my side and push again once done?

dotlambda commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25795,6 +25795,8 @@ in
 
   secretscanner = callPackage ../tools/security/secretscanner { };
 
+  semiphemeral = callPackage ../tools/social/twitter/semiphemeral { };
```
Sorry I didn't see this before.
```suggestion
  semiphemeral = callPackage ../tools/misc/semiphemeral { };
```

dotlambda reviewed (approved):
LGTM except for the path.

amanjeev commented:
thank you so much @dotlambda for your patience with me

Hoverbear reviewed (approved):
Tested on NixOS unstable and works fine.

dotlambda commented on pkgs/tools/misc/semiphemeral/default.nix:
```diff
@@ -0,0 +1,26 @@
+  pythonImportsCheck = [ "semiphemeral" ];
+
+  propagatedBuildInputs = with python3.pkgs; [ click sqlalchemy flask tweepy colorama ];
+  
```
```suggestion

```

amanjeev commented:
@Hoverbear suggested I use `nixpjgs-fmt` before committing. Will do and push! Sorry about this.
